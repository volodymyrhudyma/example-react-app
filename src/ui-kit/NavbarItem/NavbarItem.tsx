import React from 'react';
import { Wrapper, IconWrapper } from './styled/NavbarItem.styled';
import { IMenuItem } from 'src/types';
import Icon from '../Icon/Icon';

interface INavbarItemProps {
  item: IMenuItem;
  isActive: boolean;
}

const NavbarItem = ({ item, isActive }: INavbarItemProps) => (
  <Wrapper href={item.link} className={isActive ? 'active' : ''}>
    <IconWrapper>
      <Icon src={item.icon} />
    </IconWrapper>
  </Wrapper>
);

export default NavbarItem;
