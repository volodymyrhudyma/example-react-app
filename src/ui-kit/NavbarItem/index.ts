import Component from './NavbarItem';

export default Component;
