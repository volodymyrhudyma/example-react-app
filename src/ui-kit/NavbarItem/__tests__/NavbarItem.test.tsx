import React from 'react';
import { shallow } from 'enzyme';
import NavbarItem from '../NavbarItem';

describe('NavbarItem component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <NavbarItem
        isActive={false}
        item={{
          icon: 'test',
          link: '/test',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
