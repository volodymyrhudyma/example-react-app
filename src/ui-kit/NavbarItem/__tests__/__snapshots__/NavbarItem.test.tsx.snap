// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarItem component should render correctly 1`] = `
<styled.a
  className=""
  href="/test"
>
  <styled.div>
    <Icon
      alt="Icon"
      height={24}
      src="test"
      width={24}
    />
  </styled.div>
</styled.a>
`;
